public/app/features/alerting/components/PreviewQueryTab.tsx

```diff
@@ -13,14 +13,21 @@ export const PreviewQueryTab: FC<Props> = ({ data, height, width }) => {
   const [currentSeries, setSeries] = useState<number>(0);
   const theme = useTheme();
   const styles = getStyles(theme, height);
-  const series = useMemo(
-    () => data.series.map((frame, index) => ({ value: index, label: getFrameDisplayName(frame) })),
-    [data.series]
-  );
+  const series = useMemo<Array<SelectableValue<number>>>(() => {
+    if (data?.series) {
+      return data.series.map((frame, index) => ({ value: index, label: getFrameDisplayName(frame) }));
+    }
+
+    return [];
+  }, [data]);
 
   // Select padding
   const padding = 16;
 
+  if (!data?.series?.length) {
+    return <Table data={toDataFrame([])} height={height} width={width} />;
+  }
+
```

Comment:
I added this for when panelData has no series. Looks awful BTW, maybe we can enhance this experience.

Reply:
One question I have is why we're using the Table component and not the TablePanel? Wouldn't some of this be included in the later?

Reply:
I believe those have some dependencies on panel a panel?